src/PackageCompiler.jl

```diff
@@ -1363,6 +1365,38 @@ function bundle_julia_libraries(dest_dir, stdlibs)
     return
 end
 
+function bundle_julia_libexec(ctx, destdir)
+    # We only bundle the `7z` executable at the moment, which in turn is only needed by Pkg
+    @assert ctx.env.manifest !== nothing
+    has_pkg = false
+    for (_, package) in ctx.env.manifest
+        if package.name == "Pkg"
+            has_pkg = true
+            break
+        end
+    end
+    if !has_pkg
+        return
+    end
+
+    p7zip_exe = Sys.iswindows() ? "7z.exe" : "7z" 
+    bundle_libexec_dir = normpath(joinpath(dest_dir, "bin", Base.PRIVATE_LIBEXECDIR))
+    bundle_p7zip_path = joinpath(bundle_libexec_dir, p7zip_exe)
+
+    # We only bundle the `7z` from Julia
+    for julia_p7zip_path in (joinpath(Sys.BINDIR, Base.PRIVATE_LIBEXECDIR, p7zip_exe),
+                             joinpath(Sys.BINDIR, p7zip_exe))
+        if isfile(julia_p7zip_path)
+            mkdir(bundle_libexec_dir)
+            cp(julia_p7zip_path, bundle_p7zip_path)
+            return
+        end
+    end
```

Comment:
Perhaps it's just easier to use `p7zip_jll.p7zip_path`?

Reply:
I thought about this. However, in trying to remain at least somewhat portable, I decided against it and figured it might be better to just bundle whatever Julia is shipping. Otherwise, we might bundle a `7z` executable that is not usable on other machines due to some secondary dependencies that are only available on the host system. But if you think this is a non-concern, I could also use `p7zip_jll.p7zip_path`

Reply:
> I decided against it and figured it might be better to just bundle whatever Julia is shipping

`p7zip_jll.p7zip_path` _is_ what Julia is shipping. :)  JLLs that ship with Julia give you the path to the thing that ships with Julia.

Reply:
From what I understand [here](https://github.com/JuliaLang/julia/blob/d0c42840f6ff509421a3f0d28ebe5bd80c625cc3/stdlib/p7zip_jll/src/p7zip_jll.jl#L70-L81), it could also be the executable that is found through `Sys.which`. Or am I missing something?

Reply:
That is correct; but I don't expect such a Julia to be a common occurrence, and in the future if our stdlib JLLs become more JLL-like (and no longer dump things into `${prefix}/libexec` but rather into `${prefix}/share/julia/stdlib/v1.X/artifacts/1a2b3c4d/`) that path will continue to be correct, whereas the one here will no longer be correct.

Reply:
OK, gotcha. But that means I should also make `p7zip_jll` explicitly a dependency of PackageCompiler.jl, right?

Reply:
Yes

Reply:
Done in c87edb8.